modin/pandas/test/internals/test_benchmark_mode.py

```diff
@@ -11,21 +11,55 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
-# test BenchmarkMode == True
+import unittest.mock as mock
 
-from contextlib import nullcontext
 import modin.pandas as pd
 from modin.pandas.test.utils import test_data_values
-from modin.config import BenchmarkMode, StorageFormat
-from modin.test.test_utils import warns_that_defaulting_to_pandas
+from modin.config import BenchmarkMode, Engine
 
 
-def test_syncronous_mode():
+engine = Engine.get()
+
+# We have to explicitly mock subclass implementations of wait_partitions.
+if engine == "Ray":
+    wait_method = (
+        "modin.core.execution.ray.implementations.pandas_on_ray."
+        + "partitioning.partition_manager."
+        + "PandasOnRayDataframePartitionManager.wait_partitions"
+    )
+elif engine == "Dask":
+    wait_method = (
+        "modin.core.execution.dask.implementations."
+        + "pandas_on_dask.partitioning.partition_manager."
+        + "PandasOnDaskDataframePartitionManager.wait_partitions"
+    )
+else:
+    wait_method = (
+        "modin.core.dataframe.pandas.partitioning."
+        + "partition_manager.PandasDataframePartitionManager.wait_partitions"
+    )
+
+
+def test_from_environment_variable():
```

Comment:
I would suggest to make this test depending on a parameter named `value`, which get a value from the env var in `pytest.mark.parametrize` decorator. Because if someone runs the test locally without setting the env var, he gets the test failed, which might be confusing, especially for newbies.

Reply:
That's a problem that predates this PR. I'm not a newbie, but I have also been surprised to find this test failing for the reason you describe. We can open a separate issue for this.

Reply:
Yes, create one please.

Reply:
I filed #5008.

Reply:
thanks